src/daft-parquet/src/read.rs

```diff
@@ -70,18 +112,19 @@ async fn read_parquet_single(
     schema_infer_options: ParquetSchemaInferenceOptions,
     field_id_mapping: Option<Arc<BTreeMap<i32, Field>>>,
     metadata: Option<Arc<FileMetaData>>,
+    delete_rows: Option<Vec<i64>>,
 ) -> DaftResult<Table> {
     let field_id_mapping_provided = field_id_mapping.is_some();
-    let original_columns = columns;
-    let original_num_rows = num_rows;
-    let mut num_rows = num_rows;
-    let mut columns = columns.map(|s| s.iter().map(|s| s.to_string()).collect_vec());
-    let requested_columns = columns.as_ref().map(|v| v.len());
+    let columns_to_return = columns;
+    let num_rows_to_return = num_rows;
+    let mut num_rows_to_read = num_rows;
+    let mut columns_to_read = columns.map(|s| s.iter().map(|s| s.to_string()).collect_vec());
+    let requested_columns = columns_to_read.as_ref().map(|v| v.len());
     if let Some(ref pred) = predicate {
-        if num_rows.is_some() {
-            num_rows = None;
+        if num_rows_to_read.is_some() {
```

Comment:
Nit: can't we just naively set to `None`?

Reply:
I think so. Unsure why it was written like this before